src/AmqpBundle/Amqp/Producer.php

```diff
@@ -51,6 +52,17 @@ public function publishMessage($message, $flags = AMQP_NOPARAM, array $attribute
                       array_merge($this->exchangeOptions['publish_attributes'], $attributes));
 
         $routingKeys = !empty($routingKeys) ? $routingKeys : $this->exchangeOptions['routing_keys'];
+
+        if ($this->eventDispatcher) {
+            $prePublishEvent = new PrePublishEvent($message, $routingKeys, $flags, $attributes);
+
+            $this->eventDispatcher->dispatch($prePublishEvent);
+
+            if (!$prePublishEvent->canPublish()) {
```

Comment:
Not always the cas… sometimes we just would like to **modify** the message. Since `$message` is a string we cannot modify it by reference. Maybe we should call `publish` function with event data, and to allow to modify these event's data